spotify/views.py

```diff
@@ -216,21 +217,27 @@ def clone_followed_playlist(request):
 
     sp = spotipy.Spotify(auth=token)
     current_user = User.objects.get(id = request.session['user_id'])
-    original_playlist_tracks_dict = sp.playlist_tracks(request.POST['playlist_id'], fields='items(track(uri))', limit=100, offset=0, market=None)
     list_of_track_uris = []
-    for key in original_playlist_tracks_dict['items']:
-        list_of_track_uris.append(key['track']['uri'])
     new_playlist = sp.user_playlist_create(current_user.spotify_id, request.POST['playlist_name'], public=False, description='')
-    debug_print(request.POST['playlist_name'])
-    sp.user_playlist_create(current_user.spotify_id, "Best Playlist Ever!", public=False, description='')
-    sp.user_playlist_add_tracks(current_user.id, new_playlist['id'], list_of_track_uris, position=None)
+    offset = 0
+    original_playlist_tracks_dict = sp.playlist_tracks(request.POST['playlist_id'], fields='total, items(track(uri)), items(is_local)', limit=100, offset=offset, market=None)
+    #Can only add 100 songs to a playlist at a time. This loop grabs 100 at a time from the og playlist, adds it to the new playlist, and moves the offset by 100 to grab the next portion of songs
+    while offset < original_playlist_tracks_dict['total']:
```

Comment:
question about the condition for the while loop: what does that ['total'] value signify, is it the total number of tracks in the entire playlist, or is it the total number of tracks retrieved from the api? I'm assuming it represents all tracks in the playlist. So... when we get down to line 236: offset += 100, we could end up with a situation like this:

offest = 0
total tracks = 163
is 0 < 163? yes, so:
get 100 tracks
offset = 200
is 200 < 163? yes, so:
exit loop 

Does this mean we're not getting the last 63? I'm not sure if I'm reading the code correctly so maybe its not an issue

Reply:
It will grab the last 63 without issue. It chunks by 100 each time, but by the time it checks if 200 < 163, in that last iteration it added the 63 tracks already.
The way it works:
0ffset = 0
Is 0 < 163?
Add the next 100 tracks to the playlist
Offset = 100
Is 100 < 163?
Add the next 100 tracks (in this case 63) to the playlist
Offset = 200
Is 200 < 163? Get out

Reply:
And sorry, the "total" is the total number of tracks in that given playlist that is being duplicated.

Reply:
Ahhhh yes I see it now. My bad, all right good job!